subsys/sdfw_services/services/device_info/device_info_service.cddl

```diff
@@ -0,0 +1,63 @@
+;
+; Copyright (c) 2024 Nordic Semiconductor ASA
+;
+; SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+;
+
+; .. include_startingpoint_device_info_cddl_rst
+
+stat = (SUCCESS: 0) /
+	(INTERNAL_ERROR: 16781313) /
+	(UNPROGRAMMED: 16781314)
+
+entity = (UUID: 0) /
+	(TYPE: 1) /
+	(TESTIMPRINT: 2) /
+	(PARTNO: 3) /
+	(HWREVISION: 4) /
+	(PRODUCTIONREVISION: 5)
+
+; Device Info service request payload.
+read_req = (
+	1,
+	target: entity,
+)
+
+write_req = (
+	2,
+	target: entity,
+	data: [1*8 uint] ; table of at least 1 int - at most 8 int
+)
+
+; Device Info service response to read data.
+read_resp = (
+	1,
+	target: entity,
+	status: stat,
+	data: [1*8 uint] ; table of at least 1 int - at most 8 int
+)
```

Comment:
As mentioned above, I think it would be preferable if each of the returned data types had a well defined type in this schema. Currently the length of the responses is not enforced by the protocol, and the lengths are not actually defined anywhere except for in a macro internal to `device_info_service.c`.

Unless there is a good reason to read/write the individual parts of the data you could consider combining them in a single request - that might help keep the client code size low as well as reduce the IPC overhead if you need to request multiple of these in a row. I think that would be pretty reasonable since the full data would be 64 bytes total.  For example:

```
device_info = [
    uuid: bstr .size 16,
    type: uint,
    testimprint: bstr .size 32,
    partno: uint,
    hwrevision: uint,
    productionrevision: uint,
]
```

Reply:
thank you for suggesting this ! 
it required some dipper changes to the handler / service files but it is now done.